### PR TITLE
DeeplabV3 model family

### LINKED ISSUE #199: [Semantic segmentation model] Please implement DeepLabV3 model family
DeepLabV3 is a semantic segmentation model family based on [Rethinking Atrous Convolution for Semantic Image Segmentation](https://arxiv.org/abs/1706.05587) paper.

Please implement

    deeplabv3_resnet50
    deeplabv3_resnet101

:Note: those two models pretrained weights are already available in https://torch-cdn.mlverse.org/models/vision/v1/models/

### PR DESCRIPTION
**Pull request by koshtiakanksha**
Closes #199

### CODE REVIEW

File: R/models-deeplabv3.R [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+#'   "motorbike", "person", "potted plant", "sheep", "sofa", "train", "tv/monitor"
+#' )
+#'
+#' model <- model_deeplabv3_resnet50(pretrained = TRUE)
+#' model$eval()
+#' image_batch <- torch::torch_randn(1, 3, 520, 520)
+#' output <- model(image_batch)
+#' predicted_class <- output$out[1, , 260, 260]$argmax()$item() + 1
+#' class_label <- voc_classes[predicted_class]
+#' print(paste("Predicted class at (260, 260):", class_label))
```
Comment (cregouby) [inline R/models-deeplabv3.R:43]: **improvement** could we turn that into something that ends with shows a result with `tensor_image_browse()` or `tensor_image_display()`
**suggestion** `image_batch <- torch::torch_randn(1, 3, 520, 520)` is fine, I would add a comment for the end-user to change it by a jpg image of her choice.

File: tests/testthat/test-models-deeplabv3.R
Comment (cregouby) [inline tests/testthat/test-models-deeplabv3.R:1]: **todo** **missing** could we add a test segmenting one of the jpeg image in `tests/testthat/assets/class/` to ensure the mask value inferred is correct (i.e. count of the "cat" pixels is higher than count of the "dog" pixels) ?
**improvement** this would be the perfect demonstration of the usability of the (to be) included class-to-id mapping structure

**Review by cregouby — CHANGES_REQUESTED** [review]
**praise** very nice and clear addition, thanks ! 
**todo** few cosmetic changes and alignement
**todo** add a variable that ease the class to id mapping
